### PR TITLE
Support React 16

### PR DESCRIPTION
**Pull request by itssujan**
**Summary**

Support React 16 [https://github.com/facebook/draft-js/issues/1411]

- [x] Add 16.0.0 support for  in devDependencies for `react, react-dom, react-test-renderer`
- [x] Upgrade Enzyme to 3.x
- [x] Fix failing specs due to unavailable `reactComponentExpect`

**Test Plan**
 Used `npm run test-ci` to ensure all the linting, tests and flow passes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -73,9 +75,9 @@
     "gulp-util": "^3.0.6",
     "jest": "^15.1.1",
     "prettier": "^1.7.4",
-    "react": "^15.0.0-rc.1",
-    "react-dom": "^15.0.0-rc.1",
-    "react-test-renderer": "^15.5.4",
+    "react": "^15.0.0-rc.1 || ^16.0.0",
```
Comment (itssujan) [inline package.json]: Should this be `"^15.0.0-rc.1 || ^16.0.0",` or just `"^16.0.0",`

**Comment by itssujan** [comment]
@flarnie any idea why Travis keeps failing? I have the Ci passing when I run this on my machine..

**Comment by flarnie** [comment]
Interesting - I pulled your branch, started with a fresh set of `node_modules`, and ran `yarn build && yarn test` and all tests passed for me as well.

I'll rebase this but I don't expect that to fix the failing tests. Will look a little more later to see if I can offer any tips.

**Comment by itssujan** [comment]
Thanks @flarnie

**Comment by mitermayer** [comment]
Hi @itssujan,

Thanks for submitting this PR, I have rebased and fixed the builds, will land it shortly

**Comment by mitermayer** [comment]
This may take a little longer to solve, internally we use enzyme 2.x so it wont be a straight forward landing

**Comment by itssujan** [comment]
@mitermayer is thr anything I could do to help?

**Comment by mitermayer** [comment]
Hi @itssujan ,

Thank for offering help, will keep you posted if there are any questions around it. I already imported and am fixing this internally I will try to rewrite the test case on `DraftEditorContent.react-test.js` to be tested without enzyme

**Comment by qrohlf** [comment]
Are there plans to cut a release including this peerDep bump?

Right now I'm upgrading our React version and the one thing blocking that work from shipping is draft-js. If there's anything preventing a release from being cut, I'd love to help fix that.

**Pull request closed without merging**